common/src/main/java/org/broadleafcommerce/common/extensibility/jpa/copy/DirectCopyClassTransformer.java

```diff
@@ -488,6 +491,114 @@ protected XFormParams reviewConditionalDirectCopyTransforms(String convertedClas
         return addedTemplates;
     }
 
+    protected void transformIndexAnnotations(ClassFile classFile, ClassFile templateClassFile, ConstPool constantPool) throws NotFoundException {
+        List<?> templateAttributes = templateClassFile.getAttributes();
+        Iterator<?> templateItr = templateAttributes.iterator();
+        Annotation templateTable = null;
+        while (templateItr.hasNext()) {
+            Object object = templateItr.next();
+            if (AnnotationsAttribute.class.isAssignableFrom(object.getClass())) {
+                AnnotationsAttribute attr = (AnnotationsAttribute) object;
+                Annotation[] items = attr.getAnnotations();
+                for (Annotation annotation : items) {
+                    String typeName = annotation.getTypeName();
+                    if (typeName.equals(Table.class.getName())) {
+                        templateTable = annotation;
+                    }
+                }
+            }
+        }
+
+        if (templateTable != null) {
+            AnnotationsAttribute annotationsAttribute = new AnnotationsAttribute(constantPool, AnnotationsAttribute.visibleTag);
+            List<?> attributes = classFile.getAttributes();
+            Iterator<?> itr = attributes.iterator();
+            Annotation existingTable = null;
+            while (itr.hasNext()) {
+                Object object = itr.next();
+                if (AnnotationsAttribute.class.isAssignableFrom(object.getClass())) {
+                    AnnotationsAttribute attr = (AnnotationsAttribute) object;
+                    Annotation[] items = attr.getAnnotations();
+                    for (Annotation annotation : items) {
+                        String typeName = annotation.getTypeName();
+                        if (typeName.equals(Table.class.getName())) {
+                            logger.debug("Stripping out previous Table annotation at the class level - will merge into new Table");
+                            existingTable = annotation;
+                            continue;
+                        }
+                        annotationsAttribute.addAnnotation(annotation);
+                    }
+                    itr.remove();
```

Comment:
This appears to remove _all_ annotations from the `@Entity`. Seems like you should just be removing the `@Table`, right? What exactly are you removing here? Little difficult to follow (for good reason, this is difficult to work with).

Reply:
@phillipuniverse An AnnotationAttribute is a basket that holds all of the annotations. That `itr.remove()` doesn't remove the `@Table` or `@Entity` annotation, it removes all class level annotations. However the `annotationsAttribute` list holds all of those annotation to be re-added later. This piece of code was copied from how we merge `@EntityListeners`

Reply:
Got it. So it removes everything, but also keeps track of everything that it removed, except for the `@Table` annotation. That stays removed, but you hold onto it with the `existingTable` variable. Then you modify it below (or more technically, create a new one with your changes), and then add it to the `annotationsAttribute`, which gets added back as class-level annotations.

Reply:
You got it dude!